autrainer/core/scripts/preprocess_script.py

```diff
@@ -162,28 +167,27 @@ def _preprocess_datasets(self) -> None:
                     dataset=d,
                     shuffle=False,
                     num_workers=self.num_workers,
```

Comment:
`num_workers` spawns subprocesses here for fetching, however we save examples in the main process again.
Before, we had separate threads handling the loading and saving operations, as i expect the feature extraction process to be mostly IO bound. I am not sure how we could re-use the same workers for saving the features separately, i.e. in the current setup i expect the preprocessing pipeline to run a bit slower.

Reply:
Yes, I know, I think it's fine if we keep the simpler version for now and optimize later, even though it is slower. I think the penalty might not be too great in the grand scheme of things, given that preprocessing happens once. We can also optimize it in a future PR

Reply:
Sounds good, probably not too important as you say :)